### PR TITLE
Bridge 1659 - prompt user to continue doing activities

### PR DESCRIPTION
**Pull request by joshbruhin**
Bridge-1659:
- Add abstract method to appDelegate for promptUserToContinueActivities, which will show an alert urging user to continue doing activities.
- Updates to APCActivitiesVC to listen for activity completed notification. Upon this notification, reload data and check if there are more activities to do for today. If so, prompt user to continue. If user continues, start the next activity that is not yet complete.
- Refactor a few methods: didSelectRow and viewControllerToShowForCellAtIndexPath
- Made a change to this class in it’s data loading behavior. It used to reload data every time upon viewWillAppear. Now, it loads data on viewDidLoad and upon notifications - activity complete notif and other notifs such as app entered foreground, app became active and APCDayChangeNotification

### CODE REVIEW

File: APCAppCore/APCAppCore/UI/TabBarControllers/Activities/APCActivitiesViewController.m [hunk 1]
```diff
@@ -131,53 +139,46 @@ - (void) viewWillAppear: (BOOL) animated
...
                                                object: nil];
 
     // ...but that only happens every minute or so.  This lets us respond much faster.
     [[NSNotificationCenter defaultCenter] addObserver: self
-                                             selector: @selector (checkForAndMaybeRespondToSystemDateChange)
+                                             selector: @selector (handleSystemDateChangeNotification)
                                                  name: UIApplicationDidBecomeActiveNotification
                                                object: nil];
+    
+    if (self.appDelegate.promptUserToContinueActivities) {
```
Comment (EricSiegNW) [inline APCAppCore/APCAppCore/UI/TabBarControllers/Activities/APCActivitiesViewController.m]: I think we always want to be notified, so we can update the UI. Right now, if promptUser is set to false then the UI isn't updated and the activity doesn't get a green checkmark, at least when I was testing.

Comment (joshbruhin) [reply APCAppCore/APCAppCore/UI/TabBarControllers/Activities/APCActivitiesViewController.m]: Indeed. Had this test here before changing how data is loaded. Thx for the catch. PR updated

**Review by EricSiegNW — APPROVED** [review]
nice work! Not having the spinner on activities going way more often than needed is a nice side benefit.

**Review by syoung-smallwisdom — APPROVED** [review]
Looks good

**Pull request merged**